cmd/notification-sender/main.go

```diff
@@ -0,0 +1,161 @@
+			Username:     cfg.Frontend.WriterDatabase.Username,
+			Password:     cfg.Frontend.WriterDatabase.Password,
+			Name:         cfg.Frontend.WriterDatabase.Name,
+			Host:         cfg.Frontend.WriterDatabase.Host,
+			Port:         cfg.Frontend.WriterDatabase.Port,
+			MaxOpenConns: cfg.Frontend.WriterDatabase.MaxOpenConns,
+			MaxIdleConns: cfg.Frontend.WriterDatabase.MaxIdleConns,
+		}, &types.DatabaseConfig{
+			Username:     cfg.Frontend.ReaderDatabase.Username,
+			Password:     cfg.Frontend.ReaderDatabase.Password,
+			Name:         cfg.Frontend.ReaderDatabase.Name,
+			Host:         cfg.Frontend.ReaderDatabase.Host,
+			Port:         cfg.Frontend.ReaderDatabase.Port,
+			MaxOpenConns: cfg.Frontend.ReaderDatabase.MaxOpenConns,
+			MaxIdleConns: cfg.Frontend.ReaderDatabase.MaxIdleConns,
+		})
+	}()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		bt, err := db.InitBigtable(utils.Config.Bigtable.Project, utils.Config.Bigtable.Instance, fmt.Sprintf("%d", utils.Config.Chain.Config.DepositChainID), utils.Config.RedisCacheEndpoint)
+		if err != nil {
+			logrus.Fatalf("error connecting to bigtable: %v", err)
+		}
+		db.BigtableClient = bt
+	}()
+
+	if utils.Config.TieredCacheProvider == "redis" || len(utils.Config.RedisCacheEndpoint) != 0 {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			cache.MustInitTieredCache(utils.Config.RedisCacheEndpoint)
+			logrus.Infof("tiered Cache initialized, latest finalized epoch: %v", services.LatestFinalizedEpoch())
+		}()
+	}
+
+	wg.Wait()
+	if utils.Config.TieredCacheProvider == "bigtable" && len(utils.Config.RedisCacheEndpoint) == 0 {
+		cache.MustInitTieredCacheBigtable(db.BigtableClient.GetClient(), fmt.Sprintf("%d", utils.Config.Chain.Config.DepositChainID))
+		logrus.Infof("tiered Cache initialized, latest finalized epoch: %v", services.LatestFinalizedEpoch())
+	}
+
+	if utils.Config.TieredCacheProvider != "bigtable" && utils.Config.TieredCacheProvider != "redis" {
+		logrus.Fatalf("no cache provider set, pease set TierdCacheProvider (example redis, bigtable)")
+	}
+
+	defer db.ReaderDb.Close()
+	defer db.WriterDb.Close()
+	defer db.FrontendReaderDB.Close()
+	defer db.FrontendWriterDB.Close()
+	defer db.BigtableClient.Close()
+
+	if utils.Config.Metrics.Enabled {
+		go metrics.MonitorDB(db.WriterDb)
+		DBInfo := []string{
+			cfg.WriterDatabase.Username,
+			cfg.WriterDatabase.Password,
+			cfg.WriterDatabase.Host,
+			cfg.WriterDatabase.Port,
+			cfg.WriterDatabase.Name}
+		DBStr := strings.Join(DBInfo, "-")
+		frontendDBInfo := []string{
+			cfg.Frontend.WriterDatabase.Username,
+			cfg.Frontend.WriterDatabase.Password,
+			cfg.Frontend.WriterDatabase.Host,
+			cfg.Frontend.WriterDatabase.Port,
+			cfg.Frontend.WriterDatabase.Name}
+		frontendDBStr := strings.Join(frontendDBInfo, "-")
+		if DBStr != frontendDBStr {
+			go metrics.MonitorDB(db.FrontendWriterDB)
+		}
+	}
+
+	logrus.Infof("database connection established")
+
+	services.InitNotificationSender()
```

Comment:
1:1 same function as in notifications-collector, should be handled in a function. The different line can be called after, or via flag / if

Reply:
don't understand

Reply:
`cmd/notification-sender/main.go > func main()`
and 
`cmd/notification-collector/main.go > func main()`
differ only by 1 line.

So I suggest to have only 1 function for this, where the 1 different line is called outside or called via an flag.